src/plugins/sync-plugin.js

```diff
@@ -65,6 +65,57 @@ const getUserColor = (colorMapping, colors, user) => {
   return /** @type {ColorDef} */ (colorMapping.get(user))
 }
 
+class YSyncPluginState {
+  constructor (yXmlFragment, { colors, colorMapping, permanentUserData }) {
+    this.type = yXmlFragment
+    this.doc = yXmlFragment.doc
+    this.snapshot = null
+    this.prevSnapshot = null
+    this.isChangeOrigin = false
+    this.colors = colors
+    this.colorMapping = colorMapping
+    this.permanentUserData = permanentUserData
+  }
+
+  init (view) {
+    if (this.binding) {
+      return this
+    }
+    this.binding = new ProsemirrorBinding(this.type, view)
+    return this
+  }
+
+  destroy () {
+    if (this.binding) {
+      this.binding.destroy()
+    }
+    this.binding = null
```

Comment:
Mostly for consistency: Wouldn't hurt to also explicitly init `this.binding = null` in the constructor?

Reply:
Good spot, fixed.